src/app/static/src/app/apiService.ts

```diff
@@ -104,11 +103,12 @@ export class APIService<S extends string, E extends string> implements API<S, E>
     private _handleAxiosResponse(promise: Promise<AxiosResponse>) {
         return promise.then((axiosResponse: AxiosResponse) => {
             const success = axiosResponse && axiosResponse.data;
-            const data = success.data;
+            const finalResponse = this._freezeResponse ? freezer.deepFreeze(success) : success;
```

Comment:
I think this should suffice to freeze the entire response, which can then be returned, and also have its `data` passed to `_onSuccess`, which no longer needs to do its own freeze.

I've also removed manual freeze from the other actions which were doing it in modelRun and modelCalibrate.